hs_core/hs_rdf.py

```diff
@@ -0,0 +1,358 @@
+        """Generates ORI+RDF xml for this metadata"""
+        # ORE = Namespace("http://www.openarchives.org/ore/terms/")
+        g = self.get_rdf_graph()
+        # g.add((self.rdf_metadata_subject(), ORE.describes, self.rdf_subject()))
+        return g.serialize(format='hydro-xml').decode()
+
+
+class RDF_Term_MixIn(object):
+    """
+    Provides methods for serializing a django model into and rdflib triples and deserializing from an rdflib Graph
+     back to a django model.  This mixin is designed for django models that are generic relations of another django
+     model which represents a group of metadata.
+
+     Without any configuration, the rdf_triples(subject) method will add all fields on the generic relation model except
+     for 'id', 'object_id', 'content_type' will be serialized as rdf triples.  If rdf_term is defined, it will be
+     used as the predicate.  If rdf_term is not defined, one will be defined using the class name and the
+     HSTERMS namespace.
+
+     When a model requires an rdf-xml format that the default configuration will not provide, you may overwrite
+     the rdf_triples method and the ingest_rdf method.  If you overwrite one, you must overwrite both or this mixin
+     will not work correctly.
+     """
+
+    ignored_fields = ['id', 'object_id', 'content_type']
+
+    def rdf_triples(self, subject, graph):
+        """Default implementation that parses by convention."""
+        term = self.get_class_term()
+        if not term:
+            metadata_node = subject
+        else:
+            metadata_node = BNode()
+            graph.add((subject, term, metadata_node))
+        for field_term, field_value in self.get_field_terms_and_values():
+            graph.add((metadata_node, field_term, field_value))
+
+    def get_field_terms_and_values(self, extra_ignored_fields=[]):
+        """Method that returns the field terms and field values on an object"""
+        term_values = []
+        extra_ignored_fields.extend(self.ignored_fields)
+        for field in self._meta.fields:
+            if field.name in extra_ignored_fields:
+                continue
+            field_term = self.get_field_term(field.name)
+            field_value = getattr(self, field.name)
+            if field_value and field_value is not None and field_value != 'None':
+                # urls should be a URIRef term, all others should be a Literal term
+                if isinstance(field_value, str) and field_value.startswith('http'):
+                    field_value = URIRef(field_value)
+                else:
+                    field_value = Literal(field_value)
+                term_values.append((field_term, field_value))
+        return term_values
+
+    @classmethod
+    def get_class_term(cls):
+        """
+        return a term mapped by the @rdf_terms decorator or if the deocorator is not provided, create an
+        HSTERMS.{cls.__name__}
+        """
```

Comment:
Is the comment correct? From what I can see, it returns HSTERMS.{cls.__name__} attribute rather than create it. Also, a typo for decorator in the comment.

Reply:
It is correct.  HSTERMS is an rdflib namespace.  Any attribute you access on it will return the namespace with the attribute name appended to it as a URIRef